docs-src/concepts/what-is-a-codec-server.md

```diff
@@ -2,42 +2,58 @@
 id: what-is-a-codec-server
 title: What is a Codec Server?
 sidebar_label: Codec Server
-description: A Codec Server is an HTTP server that runs data from tctl or the Web UI through a Payload Codec.
+description: A Codec Server is an HTTP server that uses your custom Payload Codec to encode and decode your data remotely through endpoints.
 tags:
   - term
 ---
 
-A Codec Server is an HTTP server that runs data from [tctl](/tctl-v1) or the [Web UI](/web-ui) through a [Payload Codec](/concepts/what-is-a-data-converter#payload-codec).
+A Codec Server is an HTTP server that uses your custom [Payload Codec](/concepts/what-is-a-data-converter#payload-codec) to encode and decode your data remotely through endpoints.
 
-- [How to set up a Codec Server](/security/how-to-set-up-codec-server)
+![](/img/tctl-diagram-codec-server.svg)
 
-By default, tctl and the Web UI use the [Default Data Converter](/concepts/what-is-a-data-converter#default-data-converter) without a [Payload Codec](/concepts/what-is-a-data-converter#payload-codec).
-If you use a Payload Codec with your SDK, you may not be able to understand the Payload data displayed in the Web UI/tctl (for example, it may be encrypted or compressed).
-In order to convert the data to its original format, you can configure the Web UI/tctl to use a Codec Server that uses your Payload Codec.
+You can [create a custom Payload Codec](/prod-readiness-context/data-encryption) with your encoding logic (such as encryption and/or compression), and apply it to the data processed in your Workflows.
 
-![](/img/tctl-diagram-codec-server.svg)
+Using a custom Payload Codec in your Codec Server enables encoding and decoding data remotely through the endpoints that you expose on the Codec Server.
```

Comment:
I would highlight explicitly that the main usage is for tctl and UI. NOT for workers and programmatic workflow clients.

Reply:
I have highlighted this in the Usage section:
"When using tctl or the Web UI to perform some operations on a Workflow Execution, you can configure the exposed Codec Server endpoints to remotely encode data sent to the Temporal Server and decode data received from the Temporal Server." 
Do you think it should be specifically mentioned here also?